quickfixj-core/src/main/java/quickfix/mina/SingleThreadedEventHandlingStrategy.java

```diff
@@ -49,6 +51,39 @@
     public SingleThreadedEventHandlingStrategy(SessionConnector connector, int queueCapacity) {
         sessionConnector = connector;
         eventQueue = new LinkedBlockingQueue<>(queueCapacity);
+        queueTracker = QueueTracker.wrap(eventQueue);
+    }
+
+    public SingleThreadedEventHandlingStrategy(SessionConnector connector, int queueLowerWatermark, int queueUpperWatermark) {
+        sessionConnector = connector;
+        eventQueue = new LinkedBlockingQueue<>();
+        queueTracker = WatermarkTracker.newMulti(eventQueue, queueLowerWatermark, queueUpperWatermark,
+                evt -> evt.quickfixSession,
+                qfSession -> { // lower watermark crossed down, while reads suspended
+                    final IoSession ioSession = lookupIoSession(qfSession);
+                    if (ioSession != null && ioSession.isReadSuspended()) {
+                        ioSession.resumeRead();
+                        LOG.info("{}: inbound queue size < lower watermark ({}), socket reads resumed",
+                                qfSession.getSessionID(), queueLowerWatermark);
+                    }
+                },
+                qfSession -> { // upper watermark crossed up, while reads active
+                    final IoSession ioSession = lookupIoSession(qfSession);
+                    if (ioSession != null && !ioSession.isReadSuspended()) {
+                        ioSession.suspendRead();
+                        LOG.info("{}: inbound queue size > upper watermark ({}), socket reads suspended",
+                                qfSession.getSessionID(), queueUpperWatermark);
+                    }
+                });
+    }
+    private static IoSession lookupIoSession(Session qfSession) {
```

Comment:
There is a similar method in the other EventHandlingStrategy. Can this be moved to the super-interface maybe?

Reply:
should be possible. objections against static final methods on interfaces?